enclave-runner/src/usercalls/mod.rs

```diff
@@ -294,6 +301,44 @@ pub(crate) struct EnclaveState {
     last_fd: AtomicUsize,
     exiting: AtomicBool,
     usercall_ext: Box<UsercallExtension>,
+    work_sender: Mutex<Option<crossbeam::crossbeam_channel::Sender<Work>>>,
+}
+
+enum Work {
+    Stopped(StoppedTcs, EnclaveEntry),
+    Running(RunningTcs, tcs::Usercall<ErasedTcs>, (u64,u64), EnclaveEntry),
+}
+
+impl Work {
+    fn do_work(self, io_send_queue: &mpsc::Sender<UsercallSendData>
+    ) -> StdResult<(), mpsc::SendError<UsercallSendData>> {
+        let buf = RefCell::new([0u8; 1024]);
+        match self {
+            Work::Stopped(tcs, mode) => {
+                let state = RunningTcs {
+                    event_queue: tcs.event_queue,
+                    pending_event_set: 0,
+                    pending_events: Default::default(),
+                };
```

Comment:
It doesn't really make sense to construct this here. Also, it's kind of weird to pass `mode` around all the time. Probably do something like this:

```
struct Work {
    tcs: RunningTcs,
    entry: CoEntry
}

enum CoEntry {
    Initial(u64, u64, u64, u64, u64),
    Resume(tcs::Usercall<ErasedTcs>, (u64,u64)),
}
```

Here `RunningTcs` would contain `mode`. I'm hoping you can remove the initial parameters from `EnclaveEntry::Library` as well.